x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/fleet_server_hosts_flyout/index.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiText,
+  EuiLink,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+
+import { HostsInput } from '../hosts_input';
+import { useStartServices } from '../../../../hooks';
+
+import { useFleetServerHostsForm } from './use_fleet_server_host_form';
+
+const FLYOUT_MAX_WIDTH = 800;
+
+export interface FleetServerHostsFlyoutProps {
+  onClose: () => void;
+  fleetServerHosts: string[];
+}
+
+export const FleetServerHostsFlyout: React.FunctionComponent<FleetServerHostsFlyoutProps> = ({
+  onClose,
+  fleetServerHosts,
+}) => {
+  const { docLinks } = useStartServices();
+
+  const form = useFleetServerHostsForm(fleetServerHosts, onClose);
```

Comment:
@kpollich @joshdover I am wondering if this is good enough for separating business and presentation logic or if we should push this further and have a container component that calls that hooks and have really pure presentation component and pass this as props to this component?

Reply:
Personally I'm in favor of the approach you've already implemented here. I don't think a container component would add much value in this particular case, as it'd mostly just serve to pass the `form` data returned from this hook down into the child component. 

I was looking at the synthetics UI extensions for Fleet recently and I thought the component/hook patterns were well done. https://github.com/elastic/kibana/tree/main/x-pack/plugins/uptime/public/components/fleet_package